### PR TITLE
show course and subject detail  trainee

### PR DESCRIPTION
**Pull request by thiephn-2606-zz**
image
image
link redmid https://edu-redmine.sun-asterisk.vn/issues/39458?issue_count=28&issue_position=1&next_issue_id=39365

### CODE REVIEW

File: training_system/app/views/layouts/_header.html.erb [hunk 1]
```diff
@@ -1,28 +1,26 @@
...
-
       <ul class="navbar-nav align-items-center  ml-md-auto ">
       </ul>
       <ul class="navbar-nav align-items-center  ml-auto ml-md-0 ">
+        <li class="nav-item">
+          <span class="nav-link">
+          <i class="ni ni-single-02 text-yellow"></i>
+            <span class="nav-link-text">
+              <% if logged_in? %>
+                <%= link_to I18n.t("layouts.header.logout"), logout_path, method: :delete, class: "btn_logout" %>
```
Comment (ducth-2598) [inline training_system/app/views/layouts/_header.html.erb]:
```suggestion
                <%= link_to t("layouts.header.logout"), logout_path, method: :delete, class: "btn_logout" %>
```

File: training_system/app/views/trainee/courses/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                  <td><%= t("views.course.show.description") %></td>
+                  <td class="text-center"><%= @course.description %></td>
+                </tr>
+                <tr>
+                  <td><%= t("views.course.show.start_date") %></td>
+                  <td class="text-center"><%= @course.start_date %></td>
+                </tr>
+                <tr>
+                  <td><%= t("views.course.show.end_date") %></td>
+                  <td class="text-center"><%= @course.end_date %></td>
```
Comment (ducth-2598) [inline training_system/app/views/trainee/courses/show.html.erb]: I18n

File: training_system/app/views/trainee/courses/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                  <td><%= t("views.course.show.name") %></td>
+                  <td class="text-center"><%= @course.name %></td>
+                </tr>
+                <tr>
+                  <td><%= t("views.course.show.description") %></td>
+                  <td class="text-center"><%= @course.description %></td>
+                </tr>
+                <tr>
+                  <td><%= t("views.course.show.start_date") %></td>
+                  <td class="text-center"><%= @course.start_date %></td>
```
Comment (ducth-2598) [inline training_system/app/views/trainee/courses/show.html.erb]: I18n

**Comment by ducth-2598** [comment]
approve

**Comment by thiephn-2606** [comment]
approve

**Comment by thiephn-2606-zz** [comment]
ready

File: training_system/app/controllers/trainee/base_controller.rb [hunk 4]
```diff
@@ -1,3 +1,5 @@
 class Trainee::BaseController < ApplicationController
   layout "trainee/base"
+  before_action :logged_in_user
+  before_action :trainee_user
```
Comment (vanvtt-0952) [inline training_system/app/controllers/trainee/base_controller.rb]: `trainee_user` -> `is_trainee`

File: training_system/app/controllers/trainee/courses_controller.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  before_action :load_course, only: :show
+
+  def index
+    @course_user = current_user.courses.page(params[:page])
+                               .per(Settings.courses.per_page)
+    @courses = @course_user.search(params[:search])
+  end
+
+  def show
+    load_subjects @course
```
Comment (vanvtt-0952) [inline training_system/app/controllers/trainee/courses_controller.rb]: Em dùng action filter đi em

File: training_system/app/views/trainee/courses/_search.html.erb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+
+<div class="navbar-search navbar-search-light form-inline mr-sm-3 __web-inspector-hide-shortcut__" id="navbar-search-main">
+  <%= form_for( trainee_courses_path, method: :get) do %>
+    <div class="form-group mb-0 boder_search">
+      <div class="input-group input-group-alternative input-group-merge">
+        <div class="input-group-prepend">
+          <span class="input-group-text"><i class="fas fa-search"></i></span>
+        </div>
+          <%= text_field_tag :search, params[:search], :placeholder => "Search", :class =>"form-control input_search"%>
```
Comment (vanvtt-0952) [inline training_system/app/views/trainee/courses/_search.html.erb]: `:placeholder => ` thành `placeholder:`

I18n nữa kìa

**Comment by ducth-2598** [comment]
approve

**Comment by thiephn-2606-zz** [comment]
ready

**Pull request merged**